### PR TITLE
check-es-compat: Disallowed Eslint inline config comments to work with unminified code.

### LINKED ISSUE #100: Fails when checking code with `eslint-disable` comments of unknown rules.
When trying to determine which library causes check-es-compat to fail, it's necessary to run it on unminified code. This code can contain `eslint-disable` disabling rule unknown to `es-compat`.

```
// eslint-disable-next-line @typescript-eslint/no-explicit-any
function _eventFromIncompleteOnError(msg, url, line, column) {
...
```

This causes to fail with `Definition for rule not found`.

### PR DESCRIPTION
**Pull request by brabeji**
Fixes #100

### CODE REVIEW

**Comment by robatwilliams** [comment]
I've now removed the Node 14 build check gate, as the actual build configuration was removed previously.

**Comment by robatwilliams** [comment]
This change does remove the option for CLI users to disable this plugin. The reporter of https://github.com/robatwilliams/es-compat/issues/47 was doing this as a workaround (last sentence of the issue description). But they weren't using CLI.

I think this is fine. From a CLI tool perspective it shouldn't even be visible that ESLint is used under the hood.

File: packages/check-es-compat/bin/cli.mjs [hunk 1]
```diff
@@ -19,9 +19,10 @@ if (args.length === 0) {
 
 async function execute(files) {
   const eslint = new ESLint({
-    // Ignore any config files
+    // Ignore any config; it's for target's own linter setup rather than this tool.
     useEslintrc: false,
     ignore: false,
+    allowInlineConfig: false,
```
Comment (robatwilliams) [inline packages/check-es-compat/bin/cli.mjs:25]: Nice that you found the positive equivalent of `noInlineConfig`

**Review by robatwilliams — APPROVED** [review]

**Comment by brabeji** [comment]
> From a CLI tool perspective it shouldn't even be visible that ESLint is used under the hood.

agreed

**Comment by robatwilliams** [comment]
Released in https://github.com/robatwilliams/es-compat/releases/tag/v3.2.1

**Pull request merged**